rmw_fastrtps_shared_cpp/src/rmw_take.cpp

```diff
@@ -144,14 +162,6 @@ _take_sequence(
   auto info = static_cast<CustomSubscriberInfo *>(subscription->data);
   RCUTILS_CHECK_FOR_NULL_WITH_MSG(info, "custom subscriber info is null", return RMW_RET_ERROR);
 
-  // Limit the upper bound of reads to the number unread at the beginning.
-  // This prevents any samples that are added after the beginning of the
-  // _take_sequence call from being read.
-  auto unread_count = info->data_reader_->get_unread_count();
-  if (unread_count < count) {
-    count = unread_count;
-  }
```

Comment:
This is modifying the behavior of the method. I do not know how the API is intended to work by ROS 2, but up until now even if the user wanted to read more than the available samples, it would get the maximum number stored at the moment of calling the API and no sample added after that moment.

Reply:
This is actually guarded by the check on line 161. As soon as a take fails the loop is interrupted, mimicking this guard's behavior.

Reply:
Right now `get_unread_count` does not return the same information. Even though the behavior of the method is modified, the API is intended to return the desired samples independently of the samples being already written or being added while the take operation is being done.